src/ompl/geometric/planners/rrt/src/STRRTstar.cpp

```diff
@@ -67,7 +67,7 @@ void ompl::geometric::STRRTstar::setup()
     if (!tGoal_)
         tGoal_.reset(new ompl::NearestNeighborsLinear<Motion *>());
     tStart_->setDistanceFunction([this](const Motion *a, const Motion *b) { return distanceFunction(a, b); });
-    tGoal_->setDistanceFunction([this](const Motion *a, const Motion *b) { return distanceFunction(a, b); });
+    tGoal_->setDistanceFunction([this](const Motion *a, const Motion *b) { return distanceFunction(b, a); });
```

Comment:
I made a mistake, in NearestNeighborsLinear argument b is for the state, for which we are finding neighbours, 

https://github.com/ompl/ompl/blob/af4d6d625e15a4ed6d85255c718ce26bcd79e4cc/src/ompl/datastructures/NearestNeighborsLinear.h#L105

I fixed it locally, but strrt* does not work, with or without this fix. This means there is another problem.

Reply:
Hi @kaizer-nurik. Did you check if you can run the demo scripts of ST-RRT* in the demo folder? It is important to define a `MotionValidator` as described in the `SpaceTimePlanning.cpp` demo file, so that no negative time edges are added. If you still have issues, please don't hesitate to contact the author @frangrothe.